### PR TITLE
created FileReaderService and tests for it

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

**Review by AnaLanda — COMMENTED** [review]
Looks good) Not sure if it would be better to keep the tests as they are or to create a small file with the text content already in it and remove the logic for writing to and deleting the file in the tests.

File: src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FileReaderServiceImpl implements FileReaderService {
+    @Override
+    public List<String> read(String path) throws ReadFileException {
+        try (Stream<String> stream = Files.lines(Paths.get(path))) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java]: let's better move this out from try-with-resources
and just use try-catch

File: src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FileReaderServiceImpl implements FileReaderService {
+    @Override
+    public List<String> read(String path) throws ReadFileException {
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java]: do we need to indicate that method throws RuntimeException?

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileReaderTest {
+    private static final String PATH_TO_VALID_TEST_FILE = "src/main/resources/test_file.csv";
+    private static final String WRONG_PATH = "src/main/resources/file.csv";
+    private static final String CONTENT = "I, wont, to, find, job!";
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]:
```suggestion
    private static final String CONTENT = "I, want, to, find, job!";
```
 👍

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.IOException;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileReaderTest {
+    private static final String PATH_TO_VALID_TEST_FILE = "src/main/resources/test_file.csv";
+    private static final String WRONG_PATH = "src/main/resources/file.csv";
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]:
```suggestion
    private static final String PATH_TO_VALID_TEST_FILE = "src/test/resources/test_file.csv";
    private static final String WRONG_PATH = "src/test/resources/file.csv";
```

resources for testing should better lay here

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        File validFile = new File(PATH_TO_VALID_TEST_FILE);
+        try(FileWriter writer = new FileWriter(validFile, false))
+        {
+            writer.write(CONTENT);
+            writer.flush();
+        }
+        catch(IOException ex){
+            System.out.println(ex.getMessage());
+        }
+    }
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]:
```suggestion
    private FileReaderService service = new FileReaderServiceImpl();
    
    @BeforeClass
    public static void createTestResources() {
        File validFile = new File(PATH_TO_VALID_TEST_FILE);
        try (FileWriter writer = new FileWriter(validFile, false)) {
            writer.write(CONTENT);
            writer.flush();
        } catch(IOException ex) {
            System.out.println(ex.getMessage());
        }
    }
```
what about proper code formatting?

also, let's better rethrow this exception as RuntimeException

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    public void readingValidFile_Ok() {
+        List<String> expected = List.of(CONTENT);
+        List<String> actual = service.read(PATH_TO_VALID_TEST_FILE);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = ReadFileException.class)
+    public void exceptionWithWrongPath () {
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]: please add test cases with an empty file and file with several lines of text

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
I can see your pom.xml contains `maven-war-plugin` and `spring-boot-starter-tomcat` - they are redundant
everything we need we already have in `spring-boot-starter-web`

File: src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java]: done

File: src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/example/demo/service/impl/FileReaderServiceImpl.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 3 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 4 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 6 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 5 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    @BeforeClass
+    public static void createTestResources() {
+        File validFile = new File(PATH_TO_VALID_TEST_FILE);
+        File emptyFile = new File(PATH_TO_EMPTY_TEST_FILE);
+        try {
+            FileWriter writer = new FileWriter(validFile, false);
+            writer.write(CONTENT);
+            writer.flush();
+        } catch(IOException ex){
+            throw new RuntimeException(ex.getMessage());
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]:
```suggestion
            throw new RuntimeException(ex.getMessage(), ex);
```

File: src/test/java/com/example/demo/FileReaderTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+
+    @Test
+    public void readingValidFile_Ok() {
+        List<String> actual = service.read(PATH_TO_VALID_TEST_FILE);
+        Assert.assertEquals(CONTENT_LIST, actual);
+    }
+
+    @Test(expected = ReadFileException.class)
```
Comment (KseniiaMakarova) [inline src/test/java/com/example/demo/FileReaderTest.java]: what exception do you get here? it seems that it should not throw the exception, just read file and return the empty list?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 7 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

File: src/test/java/com/example/demo/FileReaderTest.java (hunk 8 above)
Comment (mikhail771) [reply src/test/java/com/example/demo/FileReaderTest.java]: done

**Review by KseniiaMakarova — APPROVED** [review]
